nav2_collision_monitor/src/velocity_polygon.cpp

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) 2023 Dexory
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_collision_monitor/velocity_polygon.hpp"
+
+#include "nav2_util/node_utils.hpp"
+
+namespace nav2_collision_monitor
+{
+
+VelocityPolygon::VelocityPolygon(
+  const nav2_util::LifecycleNode::WeakPtr & node, const std::string & polygon_name,
+  const std::shared_ptr<tf2_ros::Buffer> tf_buffer, const std::string & base_frame_id,
+  const tf2::Duration & transform_tolerance)
+: Polygon::Polygon(node, polygon_name, tf_buffer, base_frame_id, transform_tolerance)
+{
+  RCLCPP_INFO(logger_, "[%s]: Creating VelocityPolygon", polygon_name_.c_str());
+}
+
+VelocityPolygon::~VelocityPolygon()
+{
+  RCLCPP_INFO(logger_, "[%s]: Destroying VelocityPolygon", polygon_name_.c_str());
+}
+
+bool VelocityPolygon::getParameters(
+  std::string & /*polygon_sub_topic*/, std::string & polygon_pub_topic,
+  std::string & /*footprint_topic*/)
+{
+  auto node = node_.lock();
+  if (!node) {
+    throw std::runtime_error{"Failed to lock node"};
+  }
+
+  if (!getCommonParameters(polygon_pub_topic)) {
+    return false;
+  }
+
+  try {
+    // Get velocity_polygons parameter
+    nav2_util::declare_parameter_if_not_declared(
+      node, polygon_name_ + ".velocity_polygons", rclcpp::PARAMETER_STRING_ARRAY);
+    std::vector<std::string> velocity_polygons =
+      node->get_parameter(polygon_name_ + ".velocity_polygons").as_string_array();
+
+    // holonomic param
+    nav2_util::declare_parameter_if_not_declared(
+      node, polygon_name_ + ".holonomic", rclcpp::ParameterValue(false));
+    holonomic_ = node->get_parameter(polygon_name_ + ".holonomic").as_bool();
+
+    for (std::string velocity_polygon_name : velocity_polygons) {
+      // polygon points parameter
+      std::vector<Point> poly;
+      nav2_util::declare_parameter_if_not_declared(
+        node, polygon_name_ + "." + velocity_polygon_name + ".points", rclcpp::PARAMETER_STRING);
+      std::string poly_string =
+        node->get_parameter(polygon_name_ + "." + velocity_polygon_name + ".points").as_string();
+
+      if (!getPolygonFromString(poly_string, poly)) {
+        return false;
+      }
+
+      // linear_min param
+      double linear_min;
```

Comment:
For linear & theta min, max, no `0.0` defaults, do the same `PARAMETER_DOUBLE` that they need to be  specified. The min/max angle make sense to have defaulted to `-PI, PI` so it covers all angles unless otherwise specified, don't change those.